geode-core/src/main/java/org/apache/geode/internal/cache/GridAdvisor.java

```diff
@@ -257,6 +257,9 @@ public Set adviseProfileRemove() {
 
     private String host;
 
+    /** valid only for ControllerProfile */
+    private String internalHost;
```

Comment:
It doesn't seem correct to have a field in a parent class that only belongs to one of its child classes. Can this be moved into ControllerProfile?